src/index.ts

```diff
@@ -103,6 +99,20 @@ function readSvg(options: Options = { type: 'component' }): Plugin {
     return (!str && options.type === type) || str === type
   }
 
+  function addComponentProps(data: string): string {
+    const parts = svgRegex.exec(data)
+    if (!parts) {
+      throw new Error('Invalid SVG fuck')
+    }
+
+    const [, head, body] = parts
+    let retval = `${head} {...$$restProps}${body}`
+    if (options.transformComponent) {
+      retval = options.transformComponent(data)
+    }
+    return retval
```

Comment:
This makes no sense what so ever! Why overwrite the already transformed result with the hook result, when the hook is called with the incoming, unprocessed, data.

If the purpose is that the hook should take care of ALL processing it would be preferable to have the `if`-statement first in the function and return early.

Reply:
Now that i look at this i think my intention was to pass retval instead of data to the hook (given that restProps is probably always needed)

I'll push those fixes (this + removing the swear, sorry again) to the new branch if you are ok with it

Reply:
The problem is that `svgdom` will throw when given `<svg {...$$restProps}>` since that is not valid SVG. 

This also means that the tests you wrote actually isn't testing anything for real from what I can tell.

Reply:
> The problem is that `svgdom` will throw when given `<svg {...$$restProps}>` since that is not valid SVG.
> 
> This also means that the tests you wrote actually isn't testing anything for real from what I can tell.

It's actually testing the thing i was trying out, given an added props it does what it should do. `svgdom` it's not actually needed, i was testing that out with my use case but if the end user wants to use direct string manipulation is ok to, but i do agree with you that maybe it's better to pass the raw svg than. If you prefer an early return instead of this it's obviously totally doable.

Reply:
Btw i get that you are probably salty, and the forgotten swear surely didn't help my case but i can assure my intention is completely good. I want this feature because i think it's useful for our use case but for everyone using this library. And while an early return could be better for this use case it's not like assigning a string before cause huge performance issue. It was a mistake on my end that if i actually remembered to pass result instead of data would've been probably fixed before submitting the PR.

If you want i can fix those problems and i'd say we can start again from scratch without tension. :)

Reply:
Lol, I'm not salty at all, we're all good. We have all forgotten to remove improper strings after a development cycle. No probs.

The thing was/is that it all seemed rushed and not fully thought-through. 

I really don't care too much about performance in this case since this is a build-time plugin, but what I do care about is  logical and easily understandable code, and as the code was it made no great sense to me.

Reply:
> Lol, I'm not salty at all, we're all good. We have all forgotten to remove improper strings after a development cycle. No probs.
> 
> The thing was/is that it all seemed rushed and not fully thought-through.
> 
> I really don't care too much about performance in this case since this is a build-time plugin, but what I do care about is logical and easily understandable code, and as the code was it made no great sense to me.

It was not rushed, i can assure you. I guess the main problem was that i thought of this hook for the specific use case i had and i forgot to see the more general picture. If you are ok with it i'll pick this pr again fixing those things.

I was thinking if we could also expose some utility function (for example to spread the rest props) given that is something that people will probably want to do if they use the hook.